source/h1_encoder.c

```diff
@@ -319,6 +343,347 @@ int aws_h1_encoder_start_message(
     return AWS_OP_SUCCESS;
 }
 
+static int s_h1_encoder_process_content_length_body(struct aws_byte_buf *dst, struct aws_h1_encoder *encoder) {
+    while (true) {
```

Comment:
struct aws_byte_cursor cursor = aws_byte_cursor_from_buf(dst)

while (cursor->len) {

     //do stuff to seek the cursor, e.g. call the read functions or aws_byte_cursor_advance())
}

Reply:
I actually lifted this part from the code directly from what was previously there and didn't look at it. Will update.

Reply:
I have updated this to use the remaining buffer len in the while.